content/docs/identity-providers/oidc.mdx

```diff
@@ -0,0 +1,204 @@
+      - 8000
+    environment:
+      JWKS_ENDPOINT: https://pomerium/.well-known/pomerium/jwks.json
+```
+
+Note the following points about this configuration:
+
+- `start-dev` runs the Keycloak server in Development mode
+- The admin console username and password are both `admin`
+- The Keycloak server is assigned a network alias (`keycloak.localhost.pomerium.io`) so Pomerium can generate a sign-in redirect URL that resolves to the Keycloak container on the host machine
+
+## Set up Keycloak
+
+In your terminal, run Docker Compose:
+
+```yaml
+docker compose up
+```
+
+When Keycloak is ready, navigate to `http://0.0.0.0:8080` to access the **Administration Console:**
+
+![Access the Keycloak admin console](./img/oidc/keycloak-admin-console.png)
+
+Sign in with the admin credentials.
+
+### Create a realm
+
+Keycloak uses realms to manage sets of users, credentials, roles, and groups. When you start Keycloak, your instance comes with an admin realm called `master`.
+
+To create a new non-admin group, create a new realm in the admin dashboard:
+
+1. Select the **master** dropdown
+1. Select **Create Realm**
+1. In the **Realm name** field, enter a name
+1. Select **Create**
+
+![Create a new realm](./img/oidc/keycloak-realm.gif)
+
+### Create a user
+
+Your realm has no users at first. To add a user:
+
+1. Go to the admin console (`0.0.0.0:8080/admin`)
+1. Select **Users**
+1. Select **Add users**
+1. Enter a **Username**, a **First name**, and **Last name**
+1. Select **Create**
+
+![Create a user](./img/oidc/keycloak-create-a-user.gif)
+
+Now, assign the user a password:
+
+1. Select the **Credentials** tab
+1. Select **Set password**
+1. Enter a password and confirm it
+1. Set **Temporary** to **Off** so the user doesn't have to reset the password at the first login
+1. Select **Save**
+
+![Set the user's password](./img/oidc/keycloak-set-password.png)
+
+### Create a client
+
+Lastly, you need to create a **Client** for the realm you just created:
+
+1. Go to the admin console (`0.0.0.0:8080/admin`)
+1. Select **Clients**, then **Create client**
+1. Keep **Client type** as **OpenID Connect**
+1. In the **Client ID** field, enter a name (for example, `myclient`)
+1. In the **Authentication flow** field, select **Standard flow** and **Direct access grants**
+1. Leave **Login settings** empty, and select **Save**
+
+![Create a client](./img/oidc/keycloak-create-client.gif)
+
+Update the client you created:
+
+1. Select **Access settings**
+1. In the **Valid redirect URIs** field, enter `https://authenticate.localhost.pomerium.io/oauth2/callback`
+1. Select **Save**
+
+![Add Keycloak access settings](./img/oidc/keycloak-access-settings.png)
+
+## Set up Pomerium
+
+Create a YAML file and add the following configuration:
+
+```yaml title="config.yaml"
+authenticate_service_url: https://authenticate.localhost.pomerium.io
+
+idp_provider: oidc
+idp_client_id: 'account'
+idp_provider_url: 'http://keycloak.localhost.pomerium.io:8080/realms/MyRealm'
```

Comment:
I'd recommend we pick one realm name and use it consistently throughout this guide. It looks like the screen recording above uses the realm name 'Pomerium'.

We could update line 93 to read:
> In the **Realm name** field, enter the name "Pomerium"

and this line to read:
> `idp_provider_url: 'http://keycloak.localhost.pomerium.io:8080/realms/Pomerium'`